### PR TITLE
update to use mixdown-app plugin design

### PR DESCRIPTION
**Pull request by powpowricky**

### CODE REVIEW

File: test/tests/jsonp.js [hunk 1]
```diff
@@ -18,11 +20,13 @@ suite('Test error plugin', function() {
 
   setup(function(done) {
 
-    app.plugins.use(new JsonPlugin(namespace), {
+    var documents_plugin = new JsonPlugin({
```
Comment (panthershark) [inline test/tests/jsonp.js]: why does this say "documents_plugin?"

File: package.json [hunk 2]
```diff
@@ -22,7 +22,8 @@
   },
   "homepage": "https://github.com/mixdown/mixdown-json",
   "dependencies": {
-    "lodash": "~2.4.1"
+    "lodash": "~2.4.1",
+    "mixdown-app": "^3.0.4"
```
Comment (panthershark) [inline package.json]: mixdown-app should be a devDependency

File: test/tests/jsonp.js (hunk 1 above)
Comment (powpowricky) [inline test/tests/jsonp.js]: Sorry, forgot to update it when I was copying the setup code.

rt

File: package.json (hunk 2 above)
Comment (powpowricky) [inline package.json]: Okay, I'll keep that in mind in the future.

rt

**Comment by panthershark** [comment]
- mixdown-app should be a dev dependency since it is only required for testing.  
- pluggo needs to be removed.  
- upgrade mocha to latest in dev deps.
- fix naming of documents_plugin var in test.

**Pull request merged**